### PR TITLE
src: fixes

### LINKED ISSUE #297: Unable to list specific crate numbers and decorated weapons.
**Your issue may already be reported!**
N/A

**Describe the bug**
- The bot was unable to add listings of specific Mann Co. Supply Crate # (number).
- The bot adds non-existing Mann Co. Supply Crate #'82. 

**To Reproduce**

1. Find the SKU of specific Mann Co. Supply Crate #  (example #1), at marketplace.tf by mouse-over on the item in search and see the link appeared on the left bottom of the browser (if using Chrome).
2. Copy and paste this to bot's steam chat: 
`!add sku=5022;6;c1&min=0&max=1&intent=bank&sell.keys=2&sell.metal=20&buy.keys=1&buy.metal=10`
3. Then try for Mann Co. Supply Crate #'2. 
4. Copy and paste this:
`!add sku=5041;6;c2&min=0&max=1&intent=bank&sell.keys=1&sell.metal=20&buy.keys=0&buy.metal=10`
5. Then try for Mann Co. Supply Crate #'5. 
`!add sku=5045;6;c5&min=0&max=1&intent=bank&sell.keys=0&sell.metal=20&buy.keys=0&buy.metal=1`

**Expected behavior**
-The bot replied for the #'1 crate: 
> Added "Mann Co. Supply Crate #'82".

Crate #'82 does not exist.

-The bot replied for the #'2 crate: 
> Added "Mann Co. Supply Crate #'83".

Crate #'83 does not exist.

-The bot replied for the #'5 crate: 
> Added "Mann Co. Supply Crate #'77".

Crate #'77 does exist, but not what we wanted.

**Screenshots and logs**
No1
No2
No5

**Additional context**
- For Mann Co. Supply Munition #'82, #'83, #'84, #'85, #'90, #'91, #'92 and #'103, it is possible to be added since they all have their own defindex (5734, 5735, 5742, 5752, 5781, 5802, 5803, 5859, respectively).
- I think you should review your [node-tf2-sku](https://github.com/Nicklason/node-tf2-sku)

Version: 
v2.9.5

### PR DESCRIPTION
**Pull request by idinium96**

### CODE REVIEW

File: src/classes/Commands.ts [hunk 1]
```diff
@@ -357,20 +356,24 @@ export = class Commands {
         const adminDetails = this.bot.friends.getFriend(steamID);
 
         if (isAdmin) {
-            if (parts.length < 3) {
+            const parts = message.split(' ');
+            const steamIdAndMessage = CommandParser.removeCommand(message);
+            const steamIDRegex = /^(\d+)|(STEAM_([0-5]):([0-1]):([0-9]+))|(\[([a-zA-Z]):([0-5]):([0-9]+)(:[0-9]+)?])$/;
```
Comment (viggojonasson) [inline src/classes/Commands.ts:361]: Don't use regex. Just try catch a new instance of the SteamID class.

**Review by viggjona — CHANGES_REQUESTED** [review]
Don't use regex. Just try catch a new instance of the SteamID class.

**Comment by Nicklason** [comment]
Thank you for the pull request. I have some general feedback. It is best if you make a pull request for each fix, it makes it a lot easier to review and in turn it will be merged faster. You don't need to do it for this pull request, just do it from now on :)

**Comment by idinium96** [comment]
Noted @Nicklason 👌
Could be my final, but who knows?

File: src/schemas/pricelist-add.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
-export = {
+import jsonschema from 'jsonschema';
+
+export const addSchema: jsonschema.Schema = {
```
Comment (Nicklason) [inline src/schemas/pricelist-add.ts:3]: You don't need to export it with a name just to add types, but it is fine.

Comment (idinium96) [reply src/schemas/pricelist-add.ts:3]: useful
useful for future use.

Comment (Nicklason) [reply src/schemas/pricelist-add.ts:3]: Yes I know. My point was that you don't need to export it with a name to add types.

Comment (idinium96) [reply src/schemas/pricelist-add.ts:3]: alright. If it's okay then I will just leave it as is.

Comment (idinium96) [reply src/schemas/pricelist-add.ts:3]: oh wait, actually, if you don't do this, typescript will complain.

**Comment by idinium96** [comment]
Also @Nicklason, please apply e10bfae, bd052ff, and 03d8e02 commits to prices.tf too. Thank you.

File: src/lib/extend/item/getSKU.ts [hunk 3]
```diff
@@ -322,3 +327,95 @@ function getTarget(item: EconItem, schema: SchemaManager.Schema): number | null
...
+            'Mann Co. Supply Crate Series #45': 45,
+            'Mann Co. Supply Crate Series #55': 55,
+            'Mann Co. Supply Crate Series #59': 59,
+            'Mann Co. Supply Crate Series #77': 77
+        },
+        is5068: {
+            'Salvaged Mann Co. Supply Crate #30': 30,
+            'Salvaged Mann Co. Supply Crate #40': 40,
+            'Salvaged Mann Co. Supply Crate #50': 50
+        }
```
Comment (Nicklason) [inline src/lib/extend/item/getSKU.ts:401]: I don't really like the hardcoding, could something be done about it?

**Review by Nicklason — CHANGES_REQUESTED** [review]

File: src/lib/extend/item/getSKU.ts (hunk 3 above)
Comment (idinium96) [reply src/lib/extend/item/getSKU.ts:401]: I'll leave it to you or others. Schema doesn't have this.

**Pull request closed without merging**